app/components/common/eventModal/index.test.tsx

```diff
@@ -0,0 +1,176 @@
+/* eslint-disable object-curly-newline */
+/* eslint-disable comma-dangle */
+import { render, fireEvent, screen } from '@testing-library/react';
+import { CalEvent } from '~/utils/interfaces';
+import EventModal from '.';
+
+const mockEvent: CalEvent = {
+  id: '1',
+  title: 'Mock Event',
+  start: new Date('2023-03-30T12:00:00.000Z'),
+  end: new Date('2023-03-30T14:00:00.000Z'),
+  description: '',
+  location: '',
+  visibility: 'private',
+  attendees: ['a@b.c', 'alpha@beta.gamma'],
+};
+
+const mockUpdateEvent = jest.fn();
+const mockSetCalendarEvent = jest.fn();
+const mockSetIsOpen = jest.fn();
+
+describe('EventModal', () => {
+  test('renders EventModal component', () => {
+    const { getByPlaceholderText } = render(
+      <EventModal
+        event={mockEvent}
+        eventsList={[mockEvent]}
+        currentEvent={mockEvent}
+        setIsOpen={mockSetIsOpen}
+        updateEvent={mockUpdateEvent}
+        setCalendarEvent={mockSetCalendarEvent}
+      />
+    );
+    expect(screen.getByTestId('modal-close-btn')).toBeInTheDocument();
+    expect(screen.getByTestId('modal-title')).toHaveValue('Mock Event');
+    expect(screen.getByTestId('modal-visibility')).toHaveTextContent('Private');
+    expect(getByPlaceholderText('from-date')).toBeInTheDocument();
+    expect(getByPlaceholderText('to-date')).toBeInTheDocument();
+    expect(screen.getByTestId('email-chip-wrapper')).toHaveTextContent('a@b.c');
+    expect(screen.getByTestId('email-chip-wrapper')).toHaveTextContent('alpha@beta.gamma');
+    expect(
+      (getByPlaceholderText('Enter URL or Address for the event') as HTMLInputElement).value
+    ).toBe('');
+    expect((getByPlaceholderText('Event Description') as HTMLTextAreaElement).value).toBe('');
+    expect(screen.getByTestId('modal-save-btn')).toBeInTheDocument();
+  });
+
+  test('updates event title when title input is changed', () => {
+    render(
+      <EventModal
+        event={mockEvent}
+        eventsList={[mockEvent]}
+        currentEvent={mockEvent}
+        setIsOpen={mockSetIsOpen}
+        updateEvent={mockUpdateEvent}
+        setCalendarEvent={mockSetCalendarEvent}
+      />
+    );
+    const input = screen.getByTestId('modal-title');
+    fireEvent.change(input, { target: { value: 'updated title' } });
+    expect(input).toHaveValue('updated title');
+  });
+
+  test('should call setIsOpen with false when user clicks the Close button', () => {
+    render(
+      <EventModal
+        event={mockEvent}
+        eventsList={[mockEvent]}
+        currentEvent={mockEvent}
+        setIsOpen={mockSetIsOpen}
+        updateEvent={mockUpdateEvent}
+        setCalendarEvent={mockSetCalendarEvent}
+      />
+    );
+
+    const closeButton = screen.getByTestId('modal-close-btn');
+    fireEvent.click(closeButton);
```

Comment:
Same here we can wrap fireEvent with act()

Reply:
@DeRaowl done. Please check and update